### PR TITLE
Improve formatting of record entries with dotted fields

### PR DESCRIPTION
**Pull request by sjakobi**
Addresses #2051, part 1.

### CODE REVIEW

File: dhall/tests/format/longDottedFieldsB.dhall [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{   aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+  . bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+  . ccccccccccccccccccc = x
+,   ddddddddddddddddddddddddddddddddddddddddddddddd
+  . eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
+  . fffffffffffffffffff =
+    {- comment -}
+    y
```
Comment (sjakobi) [inline dhall/tests/format/longDottedFieldsB.dhall:8]: Desired:
```suggestion
,   ddddddddddddddddddddddddddddddddddddddddddddddd
  . eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  . fffffffffffffffffff
  = {- comment -}
    y
```

File: dhall/tests/format/recordLitComments1B.dhall [hunk 2]
```diff
@@ -1,3 +1,4 @@
 { --| a comment for y
-  y = 42
+  y
+  = 42
```
Comment (Gabriel439) [inline dhall/tests/format/recordLitComments1B.dhall:3]: It seems like this one should be formatted on one line

File: dhall/tests/format/longDottedFieldsB.dhall [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{   aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+  . bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+  . ccccccccccccccccccc
+  = x
+,   ddddddddddddddddddddddddddddddddddddddddddddddd
+  . eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
+  . fffffffffffffffffff
+  =
+    {- comment -}
+    y
```
Comment (sjakobi) [inline dhall/tests/format/longDottedFieldsB.dhall:10]: Desired:

```suggestion
,   ddddddddddddddddddddddddddddddddddddddddddddddd
  . eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  . fffffffffffffffffff
  = {- comment -}
    y
```

File: dhall/tests/format/longDottedFieldsB.dhall [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+,   ggggggggggggggggggggggggggggggggggggggggggggggg
+  . hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
+  . iiiiiiiiiiiiiiiiiii
+  =
+    ''
+    foo
+    bar
+    ''
```
Comment (sjakobi) [inline dhall/tests/format/longDottedFieldsB.dhall:18]: Desired:
```suggestion
,   ggggggggggggggggggggggggggggggggggggggggggggggg
  . hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
  . iiiiiiiiiiiiiiiiiii
  = ''
    foo
    bar
    ''
```

**Comment by sjakobi** [comment]
Unfortunately I've had a little bike accident due to which my right arm is currently in a cast - probably until Friday. If this PR should be merged sooner, I'd be happy if some one else would carry it over the line. The improvements that I pointed out on dhall/tests/format/longDottedFieldsB.dhall could possibly be left for future work.

**Comment by Gabriel439** [comment]
@sjakobi: No worries!  There is no rush.  One of the reasons we cut releases frequently is so that we don't need to care about hitting any particular release boundary since another release will be right around the corner.

**Pull request closed without merging**